### ISSUE TITLE
Get a list of all first level elements in order of a query?

### ISSUE THREAD

**Issue by Pluxx** [elboza/PhpQuery#5]
Is that even possible instead of manually picking things? so that i can basically rebuild the inner content?

**Comment by elboza** [comment]
Of course. It is possible to get a list of all first-level child elements in order from a query. You can achieve this
  by using the > child combinator in your query string. This is similar to how you would do it with CSS selectors or in
  jQuery.

  Here is an example of how you would do this with the phpQuery library. Let's say you have the following HTML structure
  from the sample.html file:

   1 <div id="pp1">
   2     <div id="pp2">
   3         <p>oooo</p>
   4     </div>
   5 </div>

  You can select the div with the id pp1 and then get all its direct children like this:

    1 <?php
    2
    3 require_once __DIR__ . '/../vendor/autoload.php';
    4 use PhpQuery\PhpQuery;
    5
    6 // Load the HTML content
    7 $page = file_get_contents('test/sample.html');
    8 $pq = new PhpQuery;
    9 $pq->load_str($page);
   10
   11 // First, query for the parent element
   12 $parent = $pq->query('#pp1');
   13
   14 if ($parent->length > 0) {
   15     // Now, query for the direct children of the parent element
   16     $children = $pq->query('> *', $parent[0]);
   17
   18     echo "Found " . $children->length . " first-level elements inside #pp1:\n";
   19
   20     // You can now iterate through the children to rebuild the inner content
   21     foreach ($children as $child) {
   22         // The outerHTML of the child
   23         echo $pq->outerHTML($child) . "\n";
   24     }
   25 }
   26
   27 ?>

  When you run this code, it will produce the following output:

   1 Found 1 first-level elements inside #pp1:
   2 <div id="pp2">
   3         <p>oooo</p>
   4     </div>

  This gives you a DOMNodeList of the first-level elements, which you can then loop through to get their HTML or text
  content, effectively allowing you to "rebuild the inner content" as you wanted. If you wanted just the content inside
  each child, you could use $pq->innerHTML($child).
